compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -175,13 +187,82 @@ TEST_F(tServiceComponent, testImmediateComponent_LifeCycle_Dynamic) // DS_TOI_51
   EXPECT_FALSE(static_cast<bool>(sRef2)) << "Service must not be available after it's dependency is removed";
 }
 
+//Function to validate lazy loading of delayed component
+bool isBundleLoaded(const std::string bundleName)
+{
+  #if defined(US_PLATFORM_WINDOWS)
+
+    HMODULE hMods[1024];
+    DWORD cbNeeded;
+
+    HANDLE hProcess = GetCurrentProcess();
+
+    auto res = EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded);
+    EXPECT_NE(res, 0) << "Enumeration failed";
+    EXPECT_GT(sizeof(hMods), cbNeeded) << "Size of array is too small to hold all module handles";
+    if ((sizeof(hMods) < cbNeeded) || res == 0)
+    {
+        return false;
+    }
+
+    TCHAR szModName[MAX_PATH];
+    std::string result;
+
+    for (unsigned int i = 0; i < (cbNeeded / sizeof(HMODULE)); i++)
+    {
+        auto file = GetModuleFileNameA(hMods[i], szModName, sizeof(szModName) / sizeof(TCHAR));
+        EXPECT_NE(file, 0) << "Failed to retrive file path";
+        result += szModName;
```

Comment:
Its not necessary to loop over every module name; once the module name is found, the work is done.